src/functions-and-arrays.js

```diff
@@ -1,41 +1,162 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers( a, b) {
+  if (a>b){
+    return a;
+  }
+  else if (a<b){
+    return b;
+  }
+  else{
+    return a;
+  }
+}
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(wordArray) {
+  if (wordArray.length === 0){
+    return null;
+  }
+  if (wordArray.length === 1){
+    return wordArray[0];
+  }
+
+  let wordLength=0;
+  let wordOcurrence;
+  for(var i=0; i<= wordArray.length-1; i++){
+    if ( wordArray[i].length > wordLength ){
+      wordLength=wordArray[i].length;
+      wordOcurrence=wordArray[i];
+    }
+  }
+  return wordOcurrence;
+}
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(arr) {
+  if (arr.length === 0){
+    return 0;
+  }
+  if (arr.length === 1){
```

Comment:
same idea here, no need to check the arr.length === 1 case separately, because you would automatically add that to your sum. 0 + n = n; so your loop would only do one iteration with only one element and the "sum" of an array of 1 number only is well... that number.